driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -648,7 +648,7 @@ public static <TExpression> Bson setWindowFields(@Nullable final TExpression par
      * @since 4.3
      */
     public static <TExpression> Bson setWindowFields(@Nullable final TExpression partitionBy, @Nullable final Bson sortBy,
-                                                     final List<WindowedComputation> output) {
+                                                     final Iterable<? extends WindowedComputation> output) {
```

Comment:
This is the only thing I wanted to change. The rest is just a consequence of this change.